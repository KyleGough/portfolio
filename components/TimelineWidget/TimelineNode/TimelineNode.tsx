import { FadeIn } from '@components/FadeIn';
import { clsx } from 'clsx';
import Image, { StaticImageData } from 'next/image';
import React from 'react';

interface TimelineNodeProps {
  align: 'left' | 'right';
  date: string;
  logo: StaticImageData;
  subtitle: string;
  title: string;
}

export const TimelineNode: React.FC<TimelineNodeProps> = ({
  title,
  subtitle,
  date,
  align,
  logo,
}) => {
  return (
    <>
      {align === 'right' && <div className="hidden md:block"></div>}
      <div
        className={clsx('flex items-center justify-end h-48', {
          'md:flex-row-reverse': align === 'right',
        })}
      >
        <FadeIn>
          <div
            className={clsx('relative w-16 md:w-24 lg:w-32 mr-8', {
              'md:ml-8 md:mr-0': align === 'right',
            })}
          >
            <Image
              src={logo.src}
              alt={title}
              width={logo.width}
              height={logo.height}
              placeholder="blur"
              blurDataURL={logo.blurDataURL}
            />
          </div>
        </FadeIn>
        <FadeIn
          className={clsx('mr-8', {
            'md:ml-8 md:mr-0 md:text-right': align === 'right',
          })}
        >
          <p className="text-3xl">{title}</p>
          <p className="text-xl">{subtitle}</p>
          <p>{date}</p>
        </FadeIn>
        <div className="h-[2px] bg-timeline w-6 max-w-[12rem] grow"></div>
      </div>
      {align === 'left' && <div className="hidden md:block"></div>}
    </>
  );
};
