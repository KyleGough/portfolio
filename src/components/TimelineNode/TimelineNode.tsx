import { clsx } from 'clsx';
import React from 'react';

import { FadeIn } from '../FadeIn';

interface TimelineNodeProps {
  title: string;
  company: string;
  date: string;
  align: 'left' | 'right';
  logo?: string;
}

export const TimelineNode: React.FC<TimelineNodeProps> = ({
  title,
  company,
  date,
  align,
  logo
}) => {
  return (
    <>
      {align === 'right' && <div className='hidden md:block'></div>}
      <div className={clsx(
        'flex items-center justify-end h-48',
        { 'md:flex-row-reverse': align === 'right' }
      )}>
        <FadeIn>
          <img className={clsx(
            'w-16 md:w-24 lg:w-32 mr-8',
            { 'md:ml-8 md:mr-0': align === 'right' }
          )}
          src={logo}
          alt={company}
          />
        </FadeIn>
        <FadeIn className={clsx(
          'mr-8',
          { 'md:ml-8 md:mr-0 md:text-right': align === 'right' }
        )}>
          <p className="text-3xl">{company}</p>
          <p className="text-xl">{title}</p>
          <p>{date}</p>
        </FadeIn>
        <div className="h-px border border-timeline w-6 max-w-[12rem] grow"></div>
      </div>
      {align === 'left' && <div className='hidden md:block'></div>}
    </>
  );
};
