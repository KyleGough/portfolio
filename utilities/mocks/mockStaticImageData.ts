const mockStaticImageData = {
  src: '/#image',
  width: 100,
  height: 100,
  blurDataURL: '/#blur',
};

export default mockStaticImageData;
