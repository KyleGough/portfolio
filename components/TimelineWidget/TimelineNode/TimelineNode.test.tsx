import mockIntersectionObserver from '@mocks/mockIntersectionObserver';
import mockStaticImageData from '@mocks/mockStaticImageData';
import { render, screen } from '@testing-library/react';
import React from 'react';

import { TimelineNode } from './TimelineNode';

describe('TimelineNode component', () => {
  beforeEach(mockIntersectionObserver);

  it('renders', () => {
    const { container } = render(
      <TimelineNode
        title="Acme"
        subtitle="CEO"
        date="January 2000 - Present"
        align="left"
        logo={mockStaticImageData}
      />
    );
    expect(screen.getByText('CEO')).toBeVisible();
    expect(screen.getByText('Acme')).toBeVisible();
    expect(screen.getByText('January 2000 - Present')).toBeVisible();
    expect(container).toMatchSnapshot();
  });
});
