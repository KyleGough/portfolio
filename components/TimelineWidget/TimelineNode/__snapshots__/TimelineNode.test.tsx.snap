// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineNode component renders 1`] = `
<div>
  <div
    class="flex items-center justify-end h-48"
  >
    <div
      class="opacity-0 transition-opacity duration-1000"
    >
      <div
        class="relative w-16 md:w-24 lg:w-32 mr-8"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27100%27%20height=%27100%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="Acme"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
    </div>
    <div
      class="mr-8 opacity-0 transition-opacity duration-1000"
    >
      <p
        class="text-3xl"
      >
        Acme
      </p>
      <p
        class="text-xl"
      >
        CEO
      </p>
      <p>
        January 2000 - Present
      </p>
    </div>
    <div
      class="h-[2px] bg-timeline w-6 max-w-[12rem] grow"
    />
  </div>
  <div
    class="hidden md:block"
  />
</div>
`;
